### PR TITLE
Add scenario test for workload version

### PR DESCRIPTION
**Pull request by sed-i**
This PR adds a scenario test around `set_workload_version`.

### CODE REVIEW

**Review by PietroPasotti — CHANGES_REQUESTED** [review]
Could you bump the scenario version to 3.0? that will have `State.status.workload_version` (and some more breaking changes)

**Comment by sed-i** [comment]
> Could you bump the scenario version to 3.0? that will have `State.status.workload_version` (and some more breaking changes)

Done

**Review by PietroPasotti — CHANGES_REQUESTED** [review]
[full disclosure] I tried doing the explicit tagging comments thing we discussed. Let's see if it works!

File: tests/scenario/test_workload_version.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            config={"routing_mode": "path"},
+            containers=self.containers,
+        )
+
+    @patch("charm.TraefikIngressCharm.external_host", PropertyMock(return_value="foo.bar"))
+    @patch("charm.TraefikIngressCharm._traefik_service_running", PropertyMock(return_value=True))
+    @patch("charm.TraefikIngressCharm.version", PropertyMock(return_value="1.2.3"))
+    def test_workload_version_is_set_on_update_status(self, *_):
+        # GIVEN an initial state without the workload version set
+        out = Context(charm_type=TraefikIngressCharm).run("start", self.state)
```
[the remark this answers is not in the document]
Comment (sed-i) [reply tests/scenario/test_workload_version.py:27]: Done in next commit.

File: tests/scenario/test_workload_version.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+from unittest.mock import PropertyMock, patch
+
+from charm import TraefikIngressCharm
+from ops.model import ActiveStatus
+from scenario import Container, Context, State
+
+
+@patch("charm.KubernetesServicePatch")
+@patch("lightkube.core.client.GenericSyncClient")
+class TestWorkloadVersion(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (sed-i) [reply tests/scenario/test_workload_version.py:14]: That's a tough one.
The entire charming ecosystem is unittest for utests and pytest for itests.

On one hand, one may argue traefik should too use unittest for utest.
On the other hand, scenario tests could be considered separately from utests, and the testing framework of choice for those is not yet established.

[the remark this answers is not in the document]
Comment (PietroPasotti) [reply tests/scenario/test_workload_version.py:14]: scenario exposes a pytest plugin, and I've not yet seen unittest + scenario in action, so yeah, still TBD but if it were up to me you know what I'd pick :)

[the remark this answers is not in the document]
Comment (sed-i) [reply tests/scenario/test_workload_version.py:14]: Thanks! Captured in https://github.com/canonical/traefik-k8s-operator/issues/185.

**Pull request merged**